core/src/main/scala/magnolia1/impl.scala

```diff
@@ -97,12 +105,17 @@ object CaseClassDerivation:
       case _: (EmptyTuple, EmptyTuple) =>
         Nil
       case _: ((l *: ltail), (p *: ptail)) =>
-        def unsafeCast(any: Any) = Option.when(any == null || (any: @unchecked).isInstanceOf[p])(any.asInstanceOf[p])
         val label = constValue[l].asInstanceOf[String]
+        val tc = new SerializableFunction0[Typeclass[p]]:
+          override def apply(): Typeclass[p] = summonInline[Typeclass[p]]
+
+        val d = new SerializableFunction0[Option[p]]:
+          private def unsafeCast(any: Any) = Option.when(any == null || (any: @unchecked).isInstanceOf[p])(any.asInstanceOf[p])
+          override def apply(): Option[p] = defaults.get(label).flatten.flatMap(d => unsafeCast(d.apply))
         paramFromMaps[Typeclass, A, p](
           label,
-          CallByNeed(summonInline[Typeclass[p]]),
-          CallByNeed.withValueEvaluator(defaults.get(label).flatten.flatMap(d => unsafeCast(d.apply))),
+          new CallByNeed(tc),
+          new CallByNeed(d, () => true),
```

Comment:
I think this makes `.withValueEvaluator` unused? Maybe we could move the comment here, as what's the purpose of the `() => true`?

Reply:
Added `by-value` API for clarity